compiler/packages/babel-plugin-react-compiler/src/Entrypoint/Options.ts

```diff
@@ -271,6 +272,14 @@ export function parsePluginOptions(obj: unknown): PluginOptions {
           parsedOptions[key] = parseTargetConfig(value);
           break;
         }
+        case 'gating': {
```

Comment:
Moved gating config parsing to BabelPlugin entrypoint -- now it's unconditional and eager like the rest of our config parsing / validation. Previously, we only tried to parse gating if we had successful compilation events.